widget/richtext.go

```diff
@@ -225,6 +225,29 @@ func (t *RichText) cachedSegmentVisual(seg RichTextSegment, offset int) fyne.Can
 	return vis
 }
 
+func (t *RichText) cleanVisualCache() {
+	if len(t.visualCache) <= len(t.Segments) {
```

Comment:
With this things may still leak once... is that OK?
(i.e. if you replace 1 or 2 or more with exactly the same number then it will expand to fit the new items, which will get removed later.
Probably a fair trade-off, just wanted to check

Reply:
I think so, what's important is that it is cleared eventually and there's an upper bound on the cache size